### ISSUE TITLE
Towards scattering in the I=3 I₃=3 channel

### ISSUE THREAD

**Issue by martin-ueding** [maowerner/subduction-code#2]
We want to do scattering of three pions, starting with maximal isospin-3-component to have the simplest of these cases. This projection code here seems to be limited to I=1 I₃=1 and L=1. Perhaps I₃ isn't fixed in the code but only in the analytic Wick contractions and therefore do not interfere with the group theoretical projection.

We also want to re-do I=2 I₃=2. It seems that we really want a code which can do all the physical channels.

From what I understand from @maowerner this code is right on the border between a throw-away prototype and something that can be refactored into a general code. I do not want to fall into the “I can easily do better than that!” trap and realize that it is not as straightforward as initially thought. Therefore I see two ways going forward:

1. Take the lessons from the existing code and rewrite most parts.

   The Pandas hierarchical multi-index that is used sounds attractive at first, but today I believe that a plain long data format together with SQL-like tools (group by, filter, …) is much easier to work with.

   In general we would have to decide how tensorial quantities shall be represented. I see (1) high dimensional arrays and (2) long data format. The former might be aided by a tensor framework like Tensor Flow or Cyclops Tensor Framework.

2. Refactor the current code into something that can do any isospin and any angular momentum, possibly even (half-integer) spin.

Both options seem to require some input from @maowerner to make use of the knowledge gained by writing the current code. There are notes that I have not started reading yet, so perhaps the existing documentation is actually sufficient.

One critical part is the Maple script that generates the irreducible representations of the octahedral group. We have to think about alternatives.

- *Maple*: This is bad because neither the university nor the institute has have a license for it. We could ask the institute to buy one, though I do not believe that the institute can buy the student version for around 150 EUR.

- *Sage*: I have been looking into alternatives a bit yesterday and found that Sage has the group in some form that one can at least get the multiplication and character table, see [this snippet](https://sagecell.sagemath.org/?z=eJyNjrFOwzAQhvdIeQdLDLWFY0pYGApLJRgQ4gGqNjrcwzVy4mBfI_z2xIGAQB3QLb_-T9_d2bb3gVgEg8oEf-yjaoGCfW_MGF9sZwldagx2GIBwXxZ5HtRquGU3_9fUOmnnybdW31l0e34tygLGDUM17K7YWXwLxOuxs7nb1WXxpCnzx2nrfT7BN7y6PLfiopbfgfGfpsphK9nmswO5lGwp-dTDVuS_-2A74ov1AQJowsAInh0uxEzyUaVn2kyUi18qJIfptDehk5LvXo8GdGLaQYwY_5ozb774qH8A9juBiQ==&lang=sage).

- *SciPy*: This has Clebsch-Gordan coefficients and Wigner-3j symbols, but not really much more.

- *Mathematica*: The university has a Mathematica campus license, so although that is license riddled as well, at least *we* can use it. You can download the Mathematica Notebook from the [Wolfram MathWorld page about the octahedral group](http://mathworld.wolfram.com/OctahedralGroup.html) and see that in Mathematica from 2004 (or rather Wolfram language) there is just `PointGroup["Oh"]`. Also it has the conjugacy classes:

    ```mathematica
    ConjugacyClasses[PointGroup["Oh"]] /. rules
    ```

    ```
    {{24}, {48}, {25, 28, 45}, {21, 4, 1}, {26, 27, 29, 34, 38, 44}, {23, 
      22, 17, 15, 11, 8}, {20, 14, 10, 5, 3, 2}, {32, 35, 39, 41, 46, 
      47}, {19, 18, 16, 13, 12, 9, 7, 6}, {30, 31, 33, 36, 37, 40, 42, 
      43}}
    ```

    But `PointGroup` is not defined in Mathematica 11.

    There are newer group theory things in Mathematica, and I played around with them. I got the octahedral group via its representation as a permutation group. I have the little groups we are interested in as well as the conjugacy classes. The irreps I did not get yet, though that should in principle be possible with Young tableaux.

    This is the notebook so far: [Octahedral_Group_Exploration.pdf](https://github.com/HISKP-LQCD/sLapH-projection/files/2723592/Octahedral_Group_Exploration.pdf)

    I'll see whether somebody answered by [Mathematica question](https://mathematica.stackexchange.com/q/188767/1507), otherwise I will take a look into the Young tableaux stuff.

    It seems that Mathematica could be a viable alternative to the Maple library.

**Comment by martin-ueding** [comment]
There is another possibility that I forgot:

- *Copying tables*: We can just copy the tables from some paper and hope that there are no mistakes in there.

**Comment by martin-ueding** [comment]
And perhaps a really sensible one:

- *Maple + Something*: The group theory of the octahedral group is independent of the physical scattering. I discussed with Markus that he could let the Maple script extract all irreps and the cartesian representation (that acts on the momentum vectors; plain 3D rotation matrices) for each group in question, namely the full octahedral group and the various little groups for P²=1, 2, and 3. P²=4 is the same group theory as P²=1.

    This way we have the hard part, the irreps, fixed and we can write some theoretically simple code that just constructs the momentum combinations for a particular physical process. All the physical details would be in that code and one does not have to touch the Maple code. In case we ever want to do P²=5, we would either have to get a Maple license or create a replacement for it.

My plan is now to write down exactly what we need in order to construct operators in a particular irrep.

@urbach, @kostrzewa, I am not sure whether you got notified of this issue. Please stop me if you think that this is a bad plan.

**Comment by maowerner** [comment]
> Perhaps I₃ isn't fixed in the code but only in the analytic Wick contractions and therefore do not interfere with the group theoretical projection.

Indeed. At least for indistinguishable particles the isospin-projection basically tells you how to (anti-)symmetrise momentum components. This can be independent of the projection although of course some bookkeeping is required.

> We could ask the institute to buy one, though I do not believe that the institute can buy the student version for around 150 EUR.

I just bought it privately. I wrote an email to the HRZ, but they told me there is no way to get it. 

> In case we ever want to do P²=5, we would either have to get a Maple license or create a replacement for it.

I am using the Bethe library [1] who in term stuck with the conventions in [2]. There is a pdf for academic use available. Using Maple to write out the irreps is basically convenient, but not a necessary dependency for the code.

[1] Comput. Phys. Commun. 171, No. 2, 119-132 (2005). 
[2] Altmann, S.; Herzig, P.: Point-group theory tables. (1994)
